src/data-access/repositories/user/knexUserRepository.js

```diff
@@ -31,12 +34,58 @@ export const userRepository = {
             })
             .update(userData)
             .returning('*');
-
         return user;
     },
+
     async remove(id) {
         await db('users')
             .where('id', id)
             .update({ isDeleted: true });
+    },
+
+    async getAllGroups() {
+        const groups = await db('groups');
+        return groups;
+    },
+
+    async getGroup(id) {
+        const [group] = await db('groups')
+            .where({ id });
+        return group;
+    },
+
+    async addGroup(groupData) {
+        const [group] = await db('groups')
+            .insert(groupData)
+            .returning('*');
+        return group;
+    },
+
+    async updateGroup(id, groupData) {
+        const [group] = await db('groups')
+            .where('id', id)
+            .update(groupData)
+            .returning('*');
+        return group;
+    },
+
+    async deleteGroup(id) {
+        await db('groups')
+            .where('id', id)
+            .del();
+    },
+
+    async addUsersToGroup(groupid, userIds) {
+        const addedUsers = await db.transaction(trx => {
+            Promise.all(userIds.map(userid =>
```

Comment:
Do it in a single call. IN this case you don't event need transaction
https://knexjs.org/guide/query-builder.html#insert

Reply:
I know, but I've used a transaction here anyway because the task description explicitly tells to do so under evaluation criteria.

Reply:
I see, now feel free to do it in the right way. It won't affect anything

Reply:
removed transaction